fdb-record-layer-core/src/test/proto/test_records_text.proto

```diff
@@ -87,3 +88,21 @@ message NestedMapDocument {
     optional int64 group = 3;
     optional string text2 = 4;
 }
+
+message Email {
+    optional int64 doc_id = 1 [(field).primary_key = true];
+    message MailHeader {
+        required string key = 1;
+        required string value = 2;
+    }
+    repeated MailHeader header = 2;
+    required string mailbox = 3;
```

Comment:
In general, we prefer `optional` to `required` for fields. Given that this is test-only, it doesn't _really_ matter, but it would be good to be consistent if possible.